src/BC-Pipe-flow.f90

```diff
@@ -0,0 +1,226 @@
+    integer (kind=MPI_OFFSET_KIND)  :: disp
+    !
+    yc = yly / two
+    zc = zlz / two
+
+    if (iscalar.ne.0) then
+        !Analytical laminar temperature profile, with nu=4.36
+        nu=real(48./11.,8)
+        do is=1,numscalar
+            do k=1,xsize(3) !MIXED-TYPE BOUNDARY CONDITION : PHI = (Tw-T)/(Tw-Tb)
+                zm=dz*real(xstart(3)-1+k-1,mytype)-zc 
+                do j=1,xsize(2)
+                    if (istret.eq.0) ym=real(j+xstart(2)-1-1,mytype)*dy-yc
+                    if (istret.ne.0) ym=yp(j+xstart(2)-1)-yc
+                    r=sqrt(ym*ym+zm*zm)
+                    do i=1,xsize(1)
+                        if (r.le.rai.and.ep1(i,j,k).eq.0) then
+                            phi1(i,j,k,is)=two*nu*(three/sixteen + r**four - r**two)
+                        else
+                            phi1(i,j,k,is)=zero
+                        endif
+                    enddo
+                enddo
+            enddo
+        enddo
+    endif
+
+    ux1=zero;uy1=zero;uz1=zero
+    if (iin.ne.0) then
+       call system_clock(count=code)
+       if (iin.eq.2) code=0
+       call random_seed(size = ii)
+       call random_seed(put = code+63946*nrank*(/ (i - 1, i = 1, ii) /))
+       call random_number(ux1)
+       call random_number(uy1)
+       call random_number(uz1)
+    endif
+    !modulation of the random noise + initial velocity profile
+    do k=1,xsize(3)
+        zm=dz*real(xstart(3)-1+k-1,mytype)-zc 
+        do j=1,xsize(2)
+            if (istret.eq.0) ym=real(j+xstart(2)-1-1,mytype)*dy-yc
+            if (istret.ne.0) ym=yp(j+xstart(2)-1)-yc
+            r=sqrt(ym*ym+zm*zm)    
+            um=exp(-twenty*r*r)
+            !Poiseuille flow
+            bxx1(j,k)=two*(one-(ym**two+zm**two)/(rai**two))
+            bxy1(j,k)=zero
+            bxz1(j,k)=zero
+            do i=1,xsize(1)
+                if (r.le.rai.and.ep1(i,j,k).eq.0) then
+                    ux1(i,j,k)=init_noise*um*(two*ux1(i,j,k)-one)+bxx1(j,k)
+                    uy1(i,j,k)=init_noise*um*(two*uy1(i,j,k)-one)+bxy1(j,k)
+                    uz1(i,j,k)=init_noise*um*(two*uz1(i,j,k)-one)+bxz1(j,k)
+                else
+                    ux1(i,j,k)=zero
+                    uy1(i,j,k)=zero
+                    uz1(i,j,k)=zero
+                endif
+            enddo
+        enddo
+    enddo
+
+#ifdef DEBG
+    if (nrank .eq. 0) print *,'# init end ok'
+#endif
+    return
+
+  end subroutine init_pipe
+  !********************************************************************
+  !
+  subroutine boundary_conditions_pipe (ux,uy,uz,phi)
+  !
+  !********************************************************************
+
+    use param
+    use variables
+    use decomp_2d
+
+    implicit none
+
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3)) :: ux,uy,uz
+    real(mytype),dimension(xsize(1),xsize(2),xsize(3),numscalar) :: phi
+
+    real(mytype),dimension(ysize(1),ysize(2),ysize(3)) :: gx
```

Comment:
Remove unused variables ?

Reply:
Thank you, it has been solves in 729ad18a8460f660222110aead0c126fa5904369 .